osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -312,6 +312,10 @@ public virtual void Save(BeatmapInfo beatmapInfo, IBeatmap beatmapContent, ISkin
 
                 beatmapInfo.MD5Hash = stream.ComputeMD5Hash();
                 beatmapInfo.Hash = stream.ComputeSHA2Hash();
+                beatmapInfo.Status = BeatmapOnlineStatus.LocallyModified;
+
+                if (setInfo.Beatmaps.All(b => b.Status == BeatmapOnlineStatus.LocallyModified))
```

Comment:
Dunno which makes more sense. Maybe this shouldn’t even exist at a set level.

Reply:
I would agree with `Any()`. Is this actually used anywhere?

Reply:
See the video. In the carousel, the set header still shows "ranked" but would become "local" if changed.

Reply:
Oh I didn't notice that in my local testing, probably because it's not what I would expect.

Reply:
To confirm you would prefer it is pessimistic as proposed? (shows “local” if any single difficulty has modifications)?

Reply:
Yes. I ended up modifying every difficult in a set and it still didn't work. Or so I thought, because I'd actually only modified every _osu!_ difficulty and there were taiko difficulties not shown. It's bad UX as it is.

Reply:
I've adjusted this, but tracking is still a bit painful. The pain for me stems from the fact that there's a set and beatmap level status (done for optimisation, ie. to allow searching for "all ranked sets" i guess). Potential paths to improve this:

- It's fine the way it is, we'll live with it for now.
- We should stop reading the `BeatmapSet` level state from API responses and just compute it locally (based on the state of all beatmaps) in a single helper method.
- We should move the "LocallyModified" concept out of the enum and keep the enum updated according to the online API, instead tracking the "locally modified" state via `LastUpdated!=null` or another new property.

Regardless of the above, give the UX another try and see how it feels now.